bgpd/bgp_debug.c

```diff
@@ -498,12 +498,13 @@ const char *bgp_notify_subcode_str(char code, char subcode)
 const char *bgp_notify_admin_message(char *buf, size_t bufsz, uint8_t *data,
 				     size_t datalen)
 {
+	*buf = 0;
```

Comment:
Shouldn't we do memset() instead?

Reply:
since we want to return a NULL string, only one char set to 0 is enough. 
and a memset is done in  "zlog_sanitize" on all the buf

Reply:
Yes, but I went quickly through all the codes calling bgp_notify_admin_message() and the message buffer (aka `buf`) is never initialized to zero.

Reply:
it is better to do it *in* the function more than *outside*. I would keep the code as Francois proposed.

Reply:
I meant doing memset() inside this helper, not outside :)